lightning/src/util/enforcing_trait_impls.rs

```diff
@@ -174,26 +197,14 @@ impl Sign for EnforcingSigner {}
 
 impl Writeable for EnforcingSigner {
 	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), Error> {
+		// NOTE: This will be de-serialized by KeysInterface::read_chan_signer
+		// and it expects to see just the InMemorySigner.  The enforcement state is managed
```

Comment:
A bit confused by this sentence... If EnforcingSigner will be deserialized by `read_chan_signer`, why will it expect to see just InMemorySigner? Maybe specify what "it" is?

Should there be a "todo" somewhere for serialization of EnforcementState?

Reply:
> A bit confused by this sentence... If EnforcingSigner will be deserialized by `read_chan_signer`, why will it expect to see just InMemorySigner? Maybe specify what "it" is?

Right, rewording this.

> 
> Should there be a "todo" somewhere for serialization of EnforcementState?

This entire file is really test utilities and not public API

Since `EnforcingSigner` and `EnforcementState` are for testing only, and `TestKeysInterface` isn't serialized in any test, then there's not much todo.  A production implementation of `KeysInterface` *should* do whatever it needs to persist whatever enforcement state they come up with.